storm-core/src/jvm/org/apache/storm/topology/CheckpointTupleForwarder.java

```diff
@@ -53,6 +53,10 @@
     private long lastTxid = Long.MIN_VALUE;
     private AnchoringOutputCollector collector;
 
+    public CheckpointTupleForwarder() {
```

Comment:
Since the IStatefulBolt is no longer an IRichBolt it cannot be passed to the base class. Makes sense to move the common functionality to a base class which both StatefulBoltExecutor and CheckpointTupleForwarder inherit to avoid the confusion. Will do the refactoring.